src/appleseed/renderer/kernel/rendering/generic/generictilerenderer.cpp

```diff
@@ -131,15 +132,15 @@ namespace
             // Retrieve tile properties.
             Tile& tile = frame.image().tile(tile_x, tile_y);
             TileStack aov_tiles = frame.aov_images().tiles(tile_x, tile_y);
-            const int tile_origin_x = static_cast<int>(frame_properties.m_tile_width * tile_x);
-            const int tile_origin_y = static_cast<int>(frame_properties.m_tile_height * tile_y);
+            const int tile_origin_x = static_cast<int>(frame_properties.m_tile_width * tile_x / ((tile_level + 1) * (tile_level + 1)));
```

Comment:
You can use `square(tile_level + 1)` for brevity (`foundation::square()` is defined in `foundation/math/scalar.h`).